pkgs/os-specific/bsd/freebsd/pkgs/drm-kmod/package.nix

```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  mkDerivation,
+  fetchFromGitHub,
+  xargs-j,
+  versionData,
+  sys,
+}:
+let
+  # Based off ports tree versions
+  reldate = lib.toIntBase10 versionData.reldate;
+  branch =
+    if reldate >= 1500008 then
+      "6.1-lts"
+    else if reldate >= 1400097 then
+      "5.15-lts"
+    else if reldate >= 1302000 then
+      "5.10-lts"
+    else
+      throw "drm-kmod not supported on FreeBSD version ${reldate}";
+
+  fetchOptions = (lib.importJSON ./versions.json).${branch};
+
+in
+mkDerivation {
+  pname = "drm-kmod";
+  version = branch;
+
+  src = fetchFromGitHub fetchOptions;
+
+  extraNativeBuildInputs = [ xargs-j ];
+
+  hardeningDisable = [
+    "pic" # generates relocations the linker can't handle
+    "stackprotector" # generates stack protection for the function generating the stack canary
+  ];
+
+  # hardeningDisable = stackprotector doesn't seem to be enough, put it in cflags too
+  NIX_CFLAGS_COMPILE = "-fno-stack-protector";
+
+  env = sys.passthru.env;
+  SYSDIR = "${sys.src}/sys";
+
+  KMODDIR = "${builtins.placeholder "out"}/kernel";
+
+  meta = with lib; {
+    description = "Linux drm driver, ported to FreeBSD";
+    platforms = lib.platforms.freebsd;
+    license = with licenses; [
+      bsd2
+      gpl2Only
+    ];
+  };
+}
```

Comment:
CI errors suggests that `nixfmt` fails. Could you format with `nixfmt-rfc-style` instead?

Reply:
```nix
Checking paths: pkgs/os-specific/bsd
pkgs/os-specific/bsd/freebsd/pkgs/drm-kmod/package.nix: not formatted
Error: nixfmt failed.
Error: Process completed with exit code 1.
```

https://github.com/NixOS/nixpkgs/pull/320475/commits/bdee1e245820428f754be393d07562e358f6dd80

Specifically this causes the error, I wonder why.

Reply:
> I am using that. Which rev of nixpkgs is new enough?

not sure if this is the answer to your question but the nixfmt used by the workflow is
https://github.com/NixOS/nixpkgs/blob/a9b494b6bcc31b6df8459b307448103412270006/.github/workflows/check-nix-format.yml#L28-L30

```
NIX_PATH="nixpkgs=https://github.com/NixOS/nixpkgs/archive/4b455dc2048f73a79eb3713f342369ff58f93e0b.tar.gz" nix-env -f '<nixpkgs>' -iAP nixfmt-rfc-style
```

Reply:
I switched to that exact rev and it seems CI is happy now.